src/pollution_model/run_medusa_2.py

```diff
@@ -28,33 +30,164 @@
 log = logging.getLogger(__name__)
 
 
+def read_roof_surface_polygons(engine: Engine) -> gpd.GeoDataFrame:
+    """
+    Read building data under polygons. Then store them into database.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    """
+    # Check if the table already exist in the database
+    if tables.check_table_exists(engine, "roof_surface_points"):
+        log.info("roof_surface_points data already exists in the database.")
+    else:
+        # Read roof surface points from outside
+        # This data has the deeplearn_matclass with roof types we need
+        roof_surface_points = gpd.read_file(
+            "/home/martinnguyen204/Digital-Twin/points_fullversion.shp"
+        )
+        # Rename
+        roof_surface_points = roof_surface_points.rename(columns={
+            'deeplearn_': 'deeplearn_matclass',
+            'deeplear_1': 'deeplearn_metalconf',
+            'dvr_matcla': 'dvr_matclass',
+            'ensemble_m': 'ensemble_matclass',
+            'ensemble_1': 'ensemble_metalconf',
+            'decay_inde': 'decay_index',
+            'deeplear_2': 'deeplearn_subclass',
+            'subclass_c': 'subclass_conf',
+            'galvanised': 'galvanised_conf',
+            'building_I': 'building_id'
+        })
+        # Remove rows of building_Id and deeplearn_subclass that are NANs
+        roof_surface_points = roof_surface_points.dropna(subset=['building_id', 'deeplearn_subclass'])
+        # Store the building_point_data to the database table
+        log.info("Adding roof_surface_points table to the database.")
```

Comment:
Either move this logging up to the start of the block, or add another log message at the start of the block. It takes a long time for the gpd.read_file call to run so this progress should be logged.

Do the same for roof_surface_polygons